contracts/halo-pool/src/state.rs

```diff
@@ -14,21 +10,17 @@ pub struct Config {
 
 pub const CONFIG: Item<Config> = Item::new("config");
 
-pub const POOL_INFO: Item<PoolInfo> = Item::new("pool_info");
-
-/// Stores the last reward time which will be updated every time when the reward is withdrawn.
-pub const LAST_REWARD_TIME: Item<u64> = Item::new("last_reward_time");
-
-// Stores the accrued token per share.
-pub const ACCRUED_TOKEN_PER_SHARE: Item<Decimal> = Item::new("accrued_token_per_share");
+/// Stores pool info of multiple phases of the same pool.
+pub const POOL_INFOS: Item<PoolInfos> = Item::new("pool_infos");
 
 /// Mappping from staker address to staker balance.
-pub const STAKERS_INFO: Map<Addr, StakerRewardAssetInfo> = Map::new("stakers_info");
+pub const STAKERS_INFO: Map<Addr, StakerInfoResponse> = Map::new("stakers_info_response");
 
 #[cw_serde]
-pub struct StakerRewardAssetInfo {
-    pub amount: Uint128,      // How many staked tokens the user has provided.
-    pub reward_debt: Uint128, // Reward debt.
+pub struct StakerInfoResponse {
```

Comment:
Mấy cái response của query này thì để sang bên file msg sẽ dễ theo dõi hơn

Reply:
Moved to msg.rs